### PR TITLE
Replace pairs getNumObjects() and getObject with foreach where possible

### PR DESCRIPTION
**Pull request by makiam**
Add getObjects() method to Scene class to conform java naming conventions
getAllObjects() remains intact for backward compatibility
Replace pairs getNumObjects() and getObject() to iterate over all scene objects with foreach where possible

### CODE REVIEW

File: ArtOfIllusion/src/artofillusion/MeshEditorWindow.java [hunk 1]
```diff
@@ -1484,11 +1484,12 @@ public boolean canAcceptAsParent(TreeElement el)
         return false;
       }
     }
-    for (int i = 0; i < theScene.getNumObjects(); i++)
+    for (ObjectInfo info: theScene.getObjects())
     {
-      ObjectInfo info = theScene.getObject(i);
       if (info.getParent() == null)
+      {
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/MeshEditorWindow.java]: Does your IDE prompt you to add these braces? They are not necessary. The Single-Line version is an established part of AOI's code style.

Comment (makiam) [inline ArtOfIllusion/src/artofillusion/MeshEditorWindow.java]: Just more readable...

File: ArtOfIllusion/src/artofillusion/CSGEditorWindow.java [hunk 2]
```diff
@@ -230,6 +231,7 @@ void deselectAllCommand()
     updateMenus();
   }
 
+  @SuppressWarnings("ResultOfObjectAllocationIgnored")
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/CSGEditorWindow.java]: Please Don't add `@SuppressWarnings` without discussing with the dev team.

File: ArtOfIllusion/src/artofillusion/object/ExternalObject.java [hunk 3]
```diff
@@ -155,7 +155,7 @@ public void reloadObject()
           theObject = foundObject.getObject();
       }
     }
-    catch (Exception ex)
+    catch (IOException ex)
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/object/ExternalObject.java]: Please don't narrow exception filters without discussing with the team.

**Pull request merged**